doc/src/manual/performance-tips.md

```diff
@@ -1507,3 +1507,89 @@ The following examples may help you interpret expressions marked as containing n
         field `data::Array{T}`. But `Array` needs the dimension `N`, too, to be a concrete type.
       * Suggestion: use concrete types like `Array{T,3}` or `Array{T,N}`, where `N` is now a parameter
         of `ArrayContainer`
+
+## [Performance of captured variable](@id man-performance-captured)
+
+Consider the following example that defines two inner functions:
+```julia
+function abmult(r::Int)
+    if r >= 0
+        f = x -> x * r
+    else
+        r = -r
+        f = x -> x * r
+    end
+    return f
+end
+```
+
+Function `abmult` returns a function `f` that multiplies its argument by
+the absolute value of `r`. The two inner functions are called
+"closures". Besides closures, inner functions are also used by the
+language for `do`-blocks and for generator expressions.
+
+This style of code presents performance challenges for the language
+for the following reason. When the parser observes occurrences of
+inner functions, it emits lower-level code in which the inner
+function is pulled out as an object living at global scope. Captured
+variables from the enclosing scope such as `r` must therefore also be moved into
+objects with global scope in order to be accessible to the inner
```

Comment:
I think this is slightly misleading --- it makes it sound like `r` becomes a global variable, which is not true. You have a good description of what actually happens below, so this paragraph doesn't seem necessary.